cpp/src/arrow/util/byte_stream_split.h

```diff
@@ -171,6 +171,381 @@ void ByteStreamSplitEncodeSSE2(const uint8_t* raw_values, const size_t num_value
 
 #endif
 
+#if defined(ARROW_HAVE_AVX2)
+template <typename T>
+void _ByteStreamSplitDecodeAVX2(const uint8_t* data, int64_t num_values, int64_t stride,
```

Comment:
style: I think it would be better to move all SIMD implementation to a new header file  called something like  byte_stream_split_simd_internal.h and move put the functions in an "internal" namespace instead of prefixing with and underscore.